src/main/java/org/apache/wicket/extensions/markup/html/tree/AbstractTree.java

```diff
@@ -264,9 +256,8 @@ public Object getModelObject()
 		}
 
 		@Override
-		public void renderHead(final IHeaderResponse response)
-		{
-			super.renderHead(response);
+		public void internalRenderHead(final HtmlHeaderContainer container) {
```

Comment:
Up to Wicket 6 we had renderHead(IHeaderResponse) for components to add their own header items, and we had renderHead(HtmlHeaderContainer) which handled calling the first method as well as collecting the header items from the component's behaviors.

AbstractTree$TreeItem in Wicket 6 overrode renderHead(HtmlHeaderContainer) to also collect the header items from the TreeItem's children to optimize render times (Not sure how much sense this makes nowadays, but I didn't want to change it).

In the transition to Wicket 7 and the move to this repo here, this was changed to renderHead(IHeaderResponse). That was wrong, because that method does _not_ handle behaviors - this is the reason that AjaxLinks didn't work, because their AjaxEventBehaviors were not called to add the appropriate header items.

The method renderHead(HtmlHeaderContainer) in Wicket 6 was renamed to internalRenderHead() in Wicket 7, since it really is internal and normally components shouldn't call or override it.

However, AbstractTree has always overridden this and actually relies on it to do its special child-rendering - basically it emulates some behavior from (I think) MarkupContainer to not just render itself but also its children.

Any component needs to have its internalRenderHead() called so its behaviors will be able to add header items, and since AbstractTree does special rendering logic for Ajax updates, it also needs to do this.